app/src/main/java/com/example/trainingintive/dog_images_feature/presentation/AlbumViewModel.kt

```diff
@@ -26,27 +30,54 @@ class AlbumViewModel @Inject constructor(
     }
 
     private fun getAllImages() {
-        repository.getAllImageUrls()
-            .subscribeOn(schedulers.io())
-            .observeOn(schedulers.ui())
-            .subscribe(
-                { _imageUrls.value = it },
-                {
-                    val errorMessageId = ErrorMessageId.getId(it as Exception)
-                    navigator.sendEvent(
-                        MainScreenEvent.Error(errorMessageId)
-                    )
-                }
-            )
+        disposables +=
+            repository.getAllImageUrls()
+                .subscribeOn(schedulers.io())
+                .observeOn(schedulers.ui())
+                .subscribe(
+                    { _imageUrls.value = it.sortedBy { it.position } },
+                    {
+                        val errorMessageId = ErrorMessageId.getId(it as Exception)
+                        navigator.sendEvent(
+                            MainScreenEvent.Error(errorMessageId)
+                        )
+                    }
+                )
     }
 
-    fun changeImagePosition(from: Int, to: Int) {}
-
     fun removeImage(index: Int) {
         val imageUrl = _imageUrls.value!![index]
-        repository.remove(imageUrl)
-            .subscribeOn(schedulers.io())
-            .observeOn(schedulers.ui())
-            .subscribe()
+        disposables +=
+            repository.remove(imageUrl)
+                .subscribeOn(schedulers.io())
+                .observeOn(schedulers.ui())
+                .subscribe()
+    }
+
+    fun changeImagePositionAndUpdateDb(from: Int, to: Int) {
+        if (shouldBeChanged()) {
+            lastChangePositionTimestamp = System.currentTimeMillis()
+            disposables +=
+                repository.changeImagePosition(_imageUrls.value!!, from, to)
+                    .subscribeOn(schedulers.io())
+                    .observeOn(schedulers.ui())
+                    .subscribe()
+        }
+    }
+
+    private fun shouldBeChanged() =
+        System.currentTimeMillis() - lastChangePositionTimestamp > 100
```

Comment:
You rather should not perform such operations in VM :( it isn't a place for this

Reply:
so repo or activity?